lib/new-syntax/parser.lr

```diff
@@ -922,9 +926,9 @@ projection_nonstar_param_list : Array<[Ast.Value, string|null]> = {
 }
 
 projection_param : [Ast.Value, string|null] = {
-    v:value => [v, null];
+    v:filter_lhs_filter_value => [v, null];
 
-    v:value 'as' alias:variable_name => [v, alias];
+    v:filter_lhs_filter_value 'as' alias:variable_name => [v, alias];
```

Comment:
This change means that `[true] of ...` will be parsed as a BooleanQuestionExpression instead of a ProjectionExpression. I think that's reasonable. Maybe a comment? This part of the grammar is quite tricky...

Btw, do we have an optimize() step to optimize the filter inside BooleanQuestionExpression? And what happens if the expression is statically true or statically false? Do we remove the boolean question or always answer yes/no?